gymnasium/spaces/box.py

```diff
@@ -335,14 +328,26 @@ def _broadcast(
 ) -> NDArray[Any]:
     """Handle infinite bounds and broadcast at the same time if needed."""
     if is_float_integer(value):
-        value = get_inf(dtype, np.sign(value)) if np.isinf(value) else value
+        if np.isinf(value) and np.dtype(dtype).kind == "i":
+            value = (
```

Comment:
`value = get_integer_infinite(dtype, np.sign(value))`
You have already checked that `np.isinf` above

Edit: If we use similar code to below, do we need the `get_integer_infinite` function?

Reply:
We haven't?

Reply:
Gotcha